src/Monitor/Tracing/Zipkin.hs

```diff
@@ -259,15 +259,25 @@ b3ToHeaderValue (B3 traceID spanID isSampled isDebug mbParentID) =
       (_ , True) -> "d"
       (True, _) -> "1"
       (False, _) -> "0"
-    required = [encodeTraceID traceID, encodeSpanID spanID, state]
+    required = [encodeZipkinTraceID traceID, encodeSpanID spanID, state]
     optional = encodeSpanID <$> maybeToList mbParentID
   in BS.intercalate "-" $ fmap T.encodeUtf8 $ required ++ optional
 
+-- | Takes into account that the first position of the b3 value is the 32 or 16 lower-hex character TraceId.
+decodeZipkinTraceID :: Text -> Maybe TraceID
+decodeZipkinTraceID txt | T.length txt == 16 = decodeTraceID $ "0000000000000000" <> txt
+                        | otherwise = decodeTraceID txt
```

Comment:
Let's keep the indentation style consistent style and consolidate the common parts:

```hs
decodeZipkinTraceID txt =
  let normalized = if T.length txt == 16 then "0000000000000000" <> txt else txt
  in decodeTraceID normalized
```

Similar below for indentation.